src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisDTOService.java

```diff
@@ -0,0 +1,160 @@
+package de.tum.in.www1.artemis.service.connectors.pyris;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingSubmission;
+import de.tum.in.www1.artemis.domain.Repository;
+import de.tum.in.www1.artemis.domain.iris.message.IrisJsonMessageContent;
+import de.tum.in.www1.artemis.domain.iris.message.IrisMessage;
+import de.tum.in.www1.artemis.domain.iris.message.IrisTextMessageContent;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.service.RepositoryService;
+import de.tum.in.www1.artemis.service.connectors.GitService;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisBuildLogEntryDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisFeedbackDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisJsonMessageContentDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisMessageContentDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisMessageDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisProgrammingExerciseDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisResultDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisSubmissionDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisTextMessageContentDTO;
+
+@Service
+@Profile("iris")
+public class PyrisDTOService {
+
+    private static final Logger log = LoggerFactory.getLogger(PyrisDTOService.class);
+
+    private final GitService gitService;
+
+    private final RepositoryService repositoryService;
+
+    public PyrisDTOService(GitService gitService, RepositoryService repositoryService) {
+        this.gitService = gitService;
+        this.repositoryService = repositoryService;
+    }
+
+    /**
+     * Convert a ProgrammingExercise to a PyrisProgrammingExerciseDTO
```

Comment:
More documentation is needed here, because this method includes git checkout operations. 

In case of retrieving git files, @pzdr7 plans to offer a way to get those files from bare repositories in LocalVC without the need to clone, pull and checkout the files in the future, which should reduce effort and improve performance